ArtifactDeckCodeDotNet/CardSetApiClient.cs

```diff
@@ -1,65 +1,130 @@
-﻿using Newtonsoft.Json;
-using System;
+﻿using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Net.Http;
 using System.Threading.Tasks;
+using Newtonsoft.Json;
 
 namespace ArtifactDeckCodeDotNet
 {
-    public class CardSetApiClient
+    public class CardSetApiClient : IDisposable
     {
         HttpClient _httpClient;
-        Dictionary<string, CardSetJsonLocation> _jsonLocations = new Dictionary<string, CardSetJsonLocation>();
-        Dictionary<string, CardSetData> _cardSetData = new Dictionary<string, CardSetData>();
+        Lazy<CardSetDataCache> _cache;
 
-        public CardSetApiClient()
+        internal static string DataCacheFile = Path.Combine(Path.GetTempPath(), "ArtifactSetDataCache.json");
+
+        public CardSetApiClient() : this(new HttpClient())
         {
-            _httpClient = new HttpClient();
         }
 
         public CardSetApiClient(HttpClient httpClient)
         {
-            _httpClient = httpClient;
+            _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+            _cache = new Lazy<CardSetDataCache>(() => new CardSetDataCache(DataCacheFile));
         }
 
-        public async Task<CardSetData> GetCardSetAsync(string setId)
+        public async Task<CardSet> GetCardSetAsync(string setId, bool forceFetch = false)
         {
-            await GetJsonLocation(setId);
+            CardSetDataCache dataCache = _cache.Value;
+            if (forceFetch ||
+                !dataCache.TryGetValue(setId, out var data) ||
+                data.ExpireTimeUtc < DateTimeOffset.Now)
```

Comment:
```suggestion
                data.ExpireTimeUtc < DateTimeOffset.UtcNow)
```

Should be UtcNow I think.